inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -376,6 +376,39 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
     }
 }
 
+void FrontEnd::processTrivialCases(const Model& model) {
+    std::unordered_map<ie::DataPtr, DataVector> ieToVpuDataVector;
+    for (const auto& data : model->datas()) {
+        const auto& origData = data->origData();
+        if (origData != nullptr) {
+            ieToVpuDataVector[origData].push_back(data);
+        }
+    }
+
+    std::vector<DataVector> trivialCases;
+    for (const auto& dataObjectsWithTheSameOrigData : ieToVpuDataVector) {
+        if (dataObjectsWithTheSameOrigData.second.size() > 1) {
+            VPU_THROW_UNLESS(dataObjectsWithTheSameOrigData.second.size() == 2,
+                             "There can't be more than two data objects connected with the same origData {}",
+                             dataObjectsWithTheSameOrigData.first->getName());
```

Comment:
Nitpick: I think it'd be nice to mention those vpu data objects that are associated with the original IE data object

Reply:
Would it be better to store it as a `std::vector<std::pair<Data, Data>>` ?

Reply:
Fixed the comment.
I think `std::vector<DataVector>` is still fine because it's easier to fill this structure, we need to traverse only once, while for the `std::vector<std::pair<Data, Data>>` we need to find two relative data objects first, which can not be done in a single traverse.

Reply:
But you already have 2 relative data objects? First traverse you fill map from IE -> VPU vector, then you check VPU vector has size == 2 and fill vector<pair>?

Reply:
At the first traverse I'm filling map from IE -> VPU vector and then I only process the vectors which has size of 2, they are already filled.

Reply:
You copy from map to vector of vectors? You can copy from map to vector of pairs?

Reply:
Well, actually yes, but I don't think it would be any better, I can do it though if you insist.

Reply:
I don't insist, but still think it's better since reflects your intention and assertion (there could be only 2 data objects) more clearly

Reply:
Yeah, but the filling process doesn't look better in this case, every time we are running through the `datas()` we should decide where to put the data inside the pair, check if the this position is not already filled, then to extract the valid pairs, we should check that both `first` and `second` are not nullptr, while for the vector we can do this by checking its size, so I suggest it to leave as is.

Reply:
So will it be okay if I implement your approach at a separate PR, so it won't stop this one to be merged before FF?

Reply:
Decided offline to do it in a separate PR.